monitoring/mock_uss/interuss_logging/routes_interuss_log.py

```
@@ -0,0 +1,83 @@
+from yaml.constructor import ConstructorError
+from loguru import logger
+
+from monitoring.mock_uss import webapp
+from monitoring.mock_uss.auth import requires_scope
+from monitoring.monitorlib.scd_automated_testing.scd_injection_api import (
+    SCOPE_SCD_QUALIFIER_INJECT,
+)
+from monitoring.monitorlib.mock_uss_interface.interuss_interaction import (
+    Interaction,
+    Issue,
+)
+
+from monitoring.mock_uss.interuss_logging.config import KEY_LOG_DIR
+
+
+@webapp.route("/mock_uss/interuss/log", methods=["GET"])
+@requires_scope([SCOPE_SCD_QUALIFIER_INJECT])
+def interuss_log() -> Tuple[str, int]:
+    """
+    Returns all the interaction logs with requests that were
+    received or initiated between 'from_time' and now
+    Eg - http:/.../mock_uss/interuss/log?from_time=2023-08-30T20:48:21.900000Z
+    """
+    from_time_param = request.args.get("from_time")
+    from_time = StringBasedDateTime(from_time_param)
+    log_path = webapp.config[KEY_LOG_DIR]
+    n = len(os.listdir(log_path))
+
+    if not os.path.exists(log_path):
+        abort(404)
+    objs = []
+    for fname in os.listdir(log_path):
+        with open(os.path.join(log_path, fname), "r") as f:
+            obj = {}
+            try:
+                for obj in yaml.full_load_all(f):
+                    interaction = ImplicitDict.parse(obj, Interaction)
+                    if (
+                        ("received_at" in interaction.query.request)
+                        and interaction.query.request.received_at.datetime
+                        > from_time.datetime
+                    ):
+                        objs.append(obj)
+                    elif (
+                        "initiated_at" in interaction.query.request
+                        and interaction.query.request.initiated_at.datetime
+                        > from_time.datetime
+                    ):
+                        objs.append(obj)
+                    else:
+                        logger.debug(
```

Do we expect there to always be either `received_at` or `initiated_at`?  If so, then this case is an error and we should surface it quickly and obviously by returning 500.  If not, under what conditions would that happen?  Would we want to include those items in the results?  If so, this isn't the right logic.  If not, it seems like there shouldn't be a need to print a debug statement.

---

There is very lass chance of that happening, I just thought of logging it in case it happens. And, yes, I ll change it to error log. We don't want it to be included in response. But, because of this if we won't send the good interactions also, then we will lose them, as they ll be deleted during teardown. At this moment I am not sure, whether I should fail it completely, or just log it.

---

We always want to be in a known state -- if we know what circumstances might cause this third case and we should simply ignore the log in that case, then we probably don't need the debug statement.  If we don't know what might cause this third state and we never expect it to happen, then we're in an unknown state when it happens and we should fail safe rather than attempt to continue hoping that the unexpected situation is not indicative of a more serious problem.  If our test system is doing things we don't expect, then we have bigger problems than capturing logs or not.

---

Resolved.